usr/share/rear/lib/bootloader-functions.sh

```diff
@@ -782,6 +782,34 @@ function make_pxelinux_config {
     echo "    append initrd=$PXE_INITRD root=/dev/ram0 vga=normal rw $KERNEL_CMDLINE $PXE_RECOVER_MODE"
     echo "say ----------------------------------------------------------"
 
+    # start with optional rear http entry if specified
+    if [[ ! -z $PXE_HTTP_URL ]] ; then    
+        case "$PXE_RECOVER_MODE" in
+        "automatic")
+            echo "say rear-automatic-http - Recover $HOSTNAME (HTTP) with auto-recover kernel option"
+            echo "label rear-automatic-http"
+            echo "MENU label ^Automatic Recover $HOSTNAME (HTTP)"
+            ;;
+        "unattended")
+            echo "say rear-unattended-http - Recover $HOSTNAME (HTTP) with unattended kernel option"
+            echo "label rear-unattended-http"
+            echo "MENU label ^Unattended Recover $HOSTNAME (HTTP)"
+            ;;
+        *)
+            echo "say rear-http - Recover $HOSTNAME (HTTP)"
+            echo "label rear-http"
+            echo "MENU label ^Recover $HOSTNAME (HTTP)"
+            ;;
+        esac
+        echo "TEXT HELP"
+        echo "Rescue image kernel $KERNEL_VERSION ${IPADDR:+on $IPADDR} $(date -R)"
+        echo "${BACKUP:+BACKUP=$BACKUP} ${OUTPUT:+OUTPUT=$OUTPUT} ${BACKUP_URL:+BACKUP_URL=$BACKUP_URL}"
+        echo "ENDTEXT"
+        echo "    kernel $PXE_HTTP_URL""/""$PXE_KERNEL"
+        echo "    append initrd=$PXE_HTTP_URL""/""$PXE_INITRD root=/dev/ram0 vga=normal rw $KERNEL_CMDLINE $PXE_RECOVER_MODE"
```

Comment:
I wonder why the special quoting `""/""` is needed
because for me it works without on command line
```
# PXE_HTTP_URL="http://pxe-over-http-srv:7777/"

# PXE_KERNEL="CLIENT/CLIENT.kernel"

# PXE_INITRD="CLIENT/CLIENT.initrd.cgz"

# echo "    kernel $PXE_HTTP_URL""/""$PXE_KERNEL"
    kernel http://pxe-over-http-srv:7777//CLIENT/CLIENT.kernel

# echo "    kernel $PXE_HTTP_URL/$PXE_KERNEL"
    kernel http://pxe-over-http-srv:7777//CLIENT/CLIENT.kernel

# echo "    append initrd=$PXE_HTTP_URL""/""$PXE_INITRD root=/dev/ram0 vga=normal rw $KERNEL_CMDLINE $PXE_RECOVER_MODE"
    append initrd=http://pxe-over-http-srv:7777//CLIENT/CLIENT.initrd.cgz root=/dev/ram0 vga=normal rw  

# echo "    append initrd=$PXE_HTTP_URL/$PXE_INITRD root=/dev/ram0 vga=normal rw $KERNEL_CMDLINE $PXE_RECOVER_MODE"
    append initrd=http://pxe-over-http-srv:7777//CLIENT/CLIENT.initrd.cgz root=/dev/ram0 vga=normal rw  
```

Reply:
well.... it is actually not needed and that's why someone else is reviewing. :+1: 
I guess I had the urge to concatenate strings missing the fact that it is just echoed.